src/utils/probe-utils.js

```diff
@@ -117,116 +119,33 @@ export const prepareForProportionPlot = (
     }
     const proportions = toProportions(counts);
     return {
-      label:
-        key === 'version'
-          ? probe.metadata[key]
-          : fullBuildIDToDate(probe.metadata[key]),
+      ...probe,
+      label: key === 'version' ? probe[key] : fullBuildIDToDate(probe[key]),
       counts,
-      version: probe.metadata.version,
+      version: probe.version,
       proportions,
-      audienceSize: probe.data[0].total_users,
+      audienceSize: probe.total_users,
     };
   });
 
-export function isScalar(payload) {
-  return payload.every(
-    (aggregation) => aggregation.metadata.metric_type === 'scalar'
-  );
-}
-
-function sortByBuildID(a, b) {
-  if (a.metadata.build_id < b.metadata.build_id) return -1;
-  if (a.metadata.build_id >= b.metadata.build_id) return 1;
-  return 0;
-}
-
-export function zipByAggregationType(payload) {
-  // returns obj
-  // keyed by aggwregation type, valued by [{data: [datum], metadata}, ...]
-  const aggTypes = new Set([]);
-  payload.forEach((aggregation) => {
-    aggregation.data.forEach((histogram) => {
-      const aggType = histogram.client_agg_type;
-      aggTypes.add(aggType);
-    });
-  });
-
-  const out = {};
-  aggTypes.forEach((a) => {
-    out[a] = [];
-  });
-
-  payload.forEach((aggregation) => {
-    const { metadata } = aggregation;
-    aggregation.data.forEach((datum) => {
-      const aggType = datum.client_agg_type;
-      out[aggType].push({ data: datum, metadata });
-    });
-  });
-
-  aggTypes.forEach((a) => {
-    out[a].sort(sortByBuildID);
-  });
-  return out;
-}
-
-export function topKBuildsPerDay(dataset, k = 2) {
-  const byBuildID = groupBy(dataset, 'label', formatBuildIDToOnlyDate);
-  const topK = Object.values(byBuildID).map((matches) => {
-    const out = matches;
-    out.sort(sortByKey('audienceSize'));
-    out.reverse();
-    return out.slice(0, k);
-  });
-  const flattened = topK.flat(2);
-  flattened.sort(sortByKey('label'));
-  return flattened;
-}
-
-export function gatherBy(payload, by) {
-  const gathered = {};
-  // get the entire set of keys.
-  payload.forEach((aggregation) => {
-    aggregation.data.forEach((entry) => {
-      const aggType = by(entry);
-      if (!(aggType in gathered)) gathered[aggType] = [];
-      gathered[aggType].push({
-        data: [entry],
-        metadata: { ...aggregation.metadata },
-      });
-    });
-  });
-  return gathered;
-}
-
-export function byKeyAndAggregation(
+export function transformResponse(
   d,
```

Comment:
I know this isn't part of the pull request, but since we're here, can we give this a more descriptive name?